lib/build-asset-map-reader.js

```diff
@@ -24,3 +24,21 @@ function readJSON(assetMap) {
 		});
 	});
 }
+
+// Retries a function that returns a promise
+// The first argument is the function
+// The second argument is an array of ms to wait in between the tries
+// Returns a function that takes the same arguments as the provided function
+function retry(fn, retries) {
+	return (...params) => fn(...params).catch(err => {
+		if(retries.length === 0) {
+			throw err;
+		}
+		let backoff = retries.shift();
```

Comment:
I know this is silly, but if you reverse `retries` upon invocation, you can use `#pop` instead of `#shift`, thus avoiding rewriting the array every time.

Reply:
...while decreasing readability immensely. I don't agree that this is a good idea.